### PR TITLE
Feature/mail endpoint

### PR DESCRIPTION
**Pull request by Josecisneros001**
### What does this PR do?
- Add loading feature to form and feedback of sending status to the user.
- Update form modal to hooks

image

- Send Using static php mail.
image
image
image

#### Related issues
#124 
#111 

#### Merge Checklist:

- [X] The branch is up-to-date (i.e. rebased) with the base branch
- [X] No code style issues (eslint and flow)

### CODE REVIEW

**Review by AuroTB — CHANGES_REQUESTED** [review]
There's a problem when contacting the server, please check that.
Overal, it looks good, maybe we can change the mail scripts for a GitHub submodule (let's talk about it later).

**Review by AuroTB — APPROVED** [review]
LGTM

File: src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
 // @flow
-import React from 'react'
+import React, { useState } from 'react'
+import ReCAPTCHA from 'react-google-recaptcha'
 import {
   Form, FormGroup, Label, Input, Row, Modal, ModalHeader, ModalBody, Button,
 } from 'reactstrap'
-import sendJoinEmail from '../../../../../scripts/apiScripts'
+import { sendJoinUsEmail } from '../../../../../scripts/apiScripts'
```
Comment (paulvazbad) [inline src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js:7]: nasty path

File: src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 import './FormsModal.css'
+import FormsSnackBar from '../../../../Shared/FormsSnackBar/FormsSnackbar'
```
Comment (paulvazbad) [inline src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js:9]: Nasty path

File: src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js [hunk 3]
```diff
@@ -18,59 +20,93 @@ type Props = {
...
+  const greeting = `Join us as ${selectedPosition.title}!`
+
+  const [name, setName] = useState('')
+  const [career, setCareer] = useState('')
+  const [semester, setSemester] = useState('1')
+  const [matricualtionNumber, setMatricualtionNumber] = useState('')
+  const [comments, setComments] = useState('')
+  const [isLoading, setIsLoading] = useState('')
+  const [open, setOpen] = useState(false)
+  const [message, setMessage] = useState('Thanks for your interest! Check your Tec email.')
```
Comment (paulvazbad) [inline src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js:37]: Shouldn't the default message be "" ? Or why is it set again in line 68?

File: src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js [hunk 4]
```diff
@@ -18,59 +20,93 @@ type Props = {
...
+      setSeverity('error')
     }
   }
 
+  const handleReCaptcha = (event: SyntheticEvent<HTMLButtonElement>) => {
+    event.preventDefault()
+    const recaptcha_ref = recaptchaRef.current
+    // $FlowFixMe[incompatible-use]
+    recaptcha_ref.reset()
+    // $FlowFixMe[incompatible-use]
```
Comment (paulvazbad) [inline src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js:101]: Remove unused comments

File: src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js [hunk 5]
```diff
@@ -18,59 +20,93 @@ type Props = {
...
+        setSeverity('error')
+        throw new Error(`Something went wrong! \n${error.message}`)
+      }).finally(() => {
+        setIsLoading(false)
+        toggle()
       })
+    } else {
+      setOpen(true)
+      setMessage('Please complete all fields')
+      setSeverity('error')
```
Comment (paulvazbad) [inline src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js:92]: I believe required field logic should be handled by the HTML 'required 'keyword instead

File: src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js [hunk 6]
```diff
@@ -18,59 +20,93 @@ type Props = {
...
+  }
+
   const getError = () => {
-    const { trySubmit } = props
     if (trySubmit && name && career && semester
       && matricualtionNumber && comments
-        && (!name.value || !career.value || !semester.value
-        || !matricualtionNumber.value || !comments.value)) {
+        && (!name || !career || !semester
+        || !matricualtionNumber || !comments)) {
```
Comment (paulvazbad) [inline src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js:109]: Same as above, use the 'required' keyword in the HTML form element if possible

File: src/components/SupportUs/SendEmail/SendEmail.js [hunk 7]
```diff
@@ -135,21 +154,39 @@ function SendEmail({ language }: Props) {
...
-      alert(`Thanks for your interest! \nCheck your Tec email \n${mailParams.message}`)
-      setName('')
-      setEmail('')
-      setPhone('')
-      setMessage('')
-      setCopy(false)
+      const mailParams = createMail(recaptchaKey)
+      setIsLoading(true)
+      sendContactUsEmail(mailParams).then((result) => {
+        if (result.status === 1) {
```
Comment (paulvazbad) [inline src/components/SupportUs/SendEmail/SendEmail.js:166]: I think the code returned should be 200 (OK)

File: src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js (hunk 4 above)
Comment (AuroTB) [reply src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js:101]: those messages unable flow error types, let me take a look at them

File: src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js (hunk 5 above)
Comment (AuroTB) [reply src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js:92]: that's outside our app scope, I'll put it as a hotfix issue

File: src/components/SupportUs/SendEmail/SendEmail.js (hunk 7 above)
Comment (AuroTB) [reply src/components/SupportUs/SendEmail/SendEmail.js:166]: @Josecisneros001 I mentioned the same in a previous revision, can you change it?

File: src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js (hunk 4 above)
Comment (AuroTB) [reply src/components/Candidates/Roles/CandidatesOpenPositions/FormsModal/FormsModal.js:101]: I´ll leave them for now

**Pull request merged**